scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -301,7 +301,7 @@ private int selectAndStartTasks(Policy currentPolicy, Map<JobId, JobDescriptor>
                     while (nodeSet != null && !nodeSet.isEmpty()) {
                         EligibleTaskDescriptor taskDescriptor = tasksToSchedule.removeFirst();
                         currentJob = ((JobDescriptorImpl) jobMap.get(taskDescriptor.getJobId())).getInternal();
-                        InternalTask internalTask = currentJob.getIHMTasks().get(taskDescriptor.getTaskId());
+                        InternalTask internalTask = ((EligibleTaskDescriptorImpl) taskDescriptor).getInternal();
```

Comment:
Why can't we move most of the methods of EligibleTaskDescriptorImpl in EligibleTaskDescriptor interface? It would prevent lots of cast like this one.

Reply:
This was done on purpose, as EligibleTaskDescriptor is a public interface, usable from the rest api or the frontend, and getInternal() returns an internal object (thus only available internally)